framework/src/modules/token/token_module.ts

```diff
@@ -30,5 +40,143 @@ export class TokenModule extends BaseModule {
 			balance: BigInt(0),
 		},
 	};
-	public transactionAssets = [new TransferAsset()];
+	public reducers = {
+		credit: async (params: Record<string, unknown>, stateStore: StateStore): Promise<void> => {
+			const { address, amount } = params;
+			if (!Buffer.isBuffer(address)) {
+				throw new Error('Address must be a buffer');
+			}
+			if (typeof amount !== 'bigint') {
+				throw new Error('Amount must be a bigint');
+			}
+			const account = await stateStore.account.getOrDefault<TokenAccount>(address);
+			account.token.balance += amount;
+			if (account.token.balance < this._minRemainingBalance) {
+				throw new Error(
+					`Remaining balance must be greater than ${this._minRemainingBalance.toString()}`,
+				);
+			}
+			stateStore.account.set(address, account);
+		},
+		debit: async (params: Record<string, unknown>, stateStore: StateStore): Promise<void> => {
+			const { address, amount } = params;
+			if (!Buffer.isBuffer(address)) {
+				throw new Error('Address must be a buffer');
+			}
+			if (typeof amount !== 'bigint') {
+				throw new Error('Amount must be a bigint');
+			}
+			const account = await stateStore.account.getOrDefault<TokenAccount>(address);
+			account.token.balance -= amount;
+			if (account.token.balance < this._minRemainingBalance) {
+				throw new Error(
+					`Remaining balance must be greater than ${this._minRemainingBalance.toString()}`,
+				);
+			}
+			stateStore.account.set(address, account);
+		},
+		getBalance: async (
+			params: Record<string, unknown>,
+			stateStore: StateStore,
+		): Promise<bigint> => {
+			const { address } = params;
+			if (!Buffer.isBuffer(address)) {
+				throw new Error('Address must be a buffer.');
+			}
+			const account = await stateStore.account.getOrDefault<TokenAccount>(address);
+			return account.token.balance;
+		},
+	};
+
+	private readonly _minRemainingBalance: bigint;
+
+	public constructor(genesisConfig: GenesisConfig) {
+		super(genesisConfig);
+		const { minRemainingBalance } = this.config;
+		if (typeof minRemainingBalance !== 'string') {
+			throw new Error('minRemainingBalance in genesisConfig must be a string.');
+		}
+		this._minRemainingBalance = BigInt(minRemainingBalance);
+	}
+	// eslint-disable-next-line @typescript-eslint/member-ordering
+	public transactionAssets = [new TransferAsset(this._minRemainingBalance)];
```

Comment:
the property should be above constructor / placed in correct order, and remove eslint ignore